### PR TITLE
RP-1739: Content-filter widget support in cdap-ui

### PR DESCRIPTION
**Pull request by neelesh-nirmal**

### CODE REVIEW

**Comment by neelesh-nirmal** [comment]
See point 5 in description @ https://deepthought.guavus.com:9443/jira/browse/RP-1739

File: cdap-ui/app/directives/widget-container/widget-content-filter/widget-content-filter.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+      templateUrl: 'widget-container/widget-content-filter/widget-content-filter.html',
+      // bindToController: true,
+      scope: {
+        disabled: '=',
+        model: '=ngModel',
+        config: '='
+      },
+      controller: function ContentFilterController($scope, myHelpers, EventPipe) {
+        'ngInject';
+
```
Comment (deepak-maheshwari-ext) [inline cdap-ui/app/directives/widget-container/widget-content-filter/widget-content-filter.js:30]: please remove unnecessary white space

File: cdap-ui/app/directives/widget-container/widget-content-filter/widget-content-filter.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        function init() {
+          try {
+            // Initialize select element options.
+            $scope.fieldOptions = [];
+            $scope.fieldOptions = $scope.config['widget-attributes'].options;
+
+            // Initial value of select element.
+            $scope.model = '*';
+
```
Comment (deepak-maheshwari-ext) [inline cdap-ui/app/directives/widget-container/widget-content-filter/widget-content-filter.js]: remove unnecessary white spaces.

File: cdap-ui/app/hydrator/controllers/create/partials/nodeconfig-ctrl.js [hunk 3]
```diff
@@ -136,7 +138,56 @@ class HydratorPlusPlusNodeConfigCtrl {
...
+      }
+    });
+
   }
+
+  shouldHideWidgetBasedOnFilter(widget) {
+    if(widget.hasOwnProperty('widget-attributes') && widget['widget-attributes'].hasOwnProperty('filters')) {
+      try {
+        // By default no need to hide.
+        let matchingFilterFound = false;
```
Comment (deepak-maheshwari-ext) [inline cdap-ui/app/hydrator/controllers/create/partials/nodeconfig-ctrl.js]: is it needed ? because at last we return false by default.

**Review by deepak-maheshwari-ext — COMMENTED** [review]
please check the comments.

File: cdap-ui/app/hydrator/controllers/create/partials/nodeconfig-ctrl.js (hunk 3 above)
Comment (neelesh-nirmal) [reply cdap-ui/app/hydrator/controllers/create/partials/nodeconfig-ctrl.js]: yes its values change later in the code.

Comment (deepak-maheshwari-ext) [reply cdap-ui/app/hydrator/controllers/create/partials/nodeconfig-ctrl.js]: it is not changed more then one place so you can directly return boolean value.

**Pull request merged**